src/System.Private.CoreLib/src/System/Runtime/Bootstrap/BootstrapUTF8Encoding.cs

```diff
@@ -0,0 +1,947 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.Text;
+
+namespace System.Runtime.Bootstrap
+{
+    // This code is primarily copied from UTF8Encoding.cs' GetCharCount and GetChars
```

Comment:
You should also add comment here on why it is needed - why regular UTF8Encoding.cs cannot be used.